### PR TITLE
Include port numbers in ServiceImports

### LINKED ISSUE #385: In ServiceImports include port number(s) of the services
**What would you like to be added**:
Currently, ServiceImports populated by submariner only includes IP addresses of the services.
However it would be good to include even the Port number(s) of services.
Port numbers are already part of the ServiceImport objecgt.

**Why is this needed**:
I am trying to see if we can somehow integrate OSM (openservice-mesh by Microsoft) with Submariner.
OSM `learns' about k8s services and programs envoy. Here, k8s is a `provider' and it provides a set of services (ServiceName, IP, Ports) to OSM within the cluster.

I was thinking of Submariner as another `provider' and can feed similar information to OSM about remote clusters.
So that services across clusters can seamlessly work.

### PR DESCRIPTION
**Pull request by vthapar**
Fixes #385

Signed-off-by: Vishal Thapar <5137689+vthapar@users.noreply.github.com>

### CODE REVIEW

**Review by aswinsuryan — COMMENTED** [review]
LGTM

File: pkg/agent/controller/agent.go [hunk 1]
```diff
@@ -422,27 +423,35 @@ func (a *Controller) newServiceImport(svcExport *mcsv1a1.ServiceExport) *mcsv1a1
 	}
 }
 
-func (a *Controller) getIPsForService(service *corev1.Service, siType mcsv1a1.ServiceImportType) ([]string, error) {
+func (a *Controller) getIPsAndPortsForService(service *corev1.Service, siType mcsv1a1.ServiceImportType) (
+	[]string, []mcsv1a1.ServicePort, error) {
+	mcsPort := mcsv1a1.ServicePort{}
+	if service.Spec.Ports != nil {
```
Comment (tpantelis) [inline pkg/agent/controller/agent.go:429]:
```suggestion
	if len(service.Spec.Ports) > 0 {
```

**Review by tpantelis — CHANGES_REQUESTED** [review]

**Review by dfarrell07 — APPROVED** [review]

**Comment by tpantelis** [comment]
We should add a unit test case for this (w/ and w/o a port).  @vthapar I can do this in a followup if you want.

**Comment by vthapar** [comment]
> We should add a unit test case for this (w/ and w/o a port). @vthapar I can do this in a followup if you want.

That would be great. I was planning to do follow up but after I'm done with the submariner serviceexport changes.

**Comment by tpantelis** [comment]
> > We should add a unit test case for this (w/ and w/o a port). @vthapar I can do this in a followup if you want.
> 
> That would be great. I was planning to do follow up but after I'm done with the submariner serviceexport changes.

OK I'll followup with UTs.

**Review by tpantelis — APPROVED** [review]

**Comment by dfarrell07** [comment]
This was backported in https://github.com/submariner-io/lighthouse/pull/450.

**Pull request merged**